jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java

```diff
@@ -157,6 +157,9 @@ public void restarted(String step, Throwable cause) {
 
     @Override
     protected String format(String key, String defaultPattern, Object... args) {
+        if ("beforeExampleParameters".equals(key)) {
+            scenarioCompleted = false;
```

Comment:
not really clear: why is `"beforeExampleParameters"` event resets scenario state?